toolchain/check/check.cpp

```diff
@@ -61,16 +61,39 @@ static auto RenderImportKey(ImportKey import_key) -> std::string {
 static auto TrackImport(Map<ImportKey, UnitAndImports*>& api_map,
                         Map<ImportKey, Parse::NodeId>* explicit_import_map,
                         UnitAndImports& unit_info,
-                        Parse::Tree::PackagingNames import) -> void {
+                        Parse::Tree::PackagingNames import, bool fuzzing)
+    -> void {
   const auto& packaging = unit_info.parse_tree().packaging_decl();
 
   IdentifierId file_package_id =
       packaging ? packaging->names.package_id : IdentifierId::Invalid;
-  auto import_key = GetImportKey(unit_info, file_package_id, import);
+  const auto import_key = GetImportKey(unit_info, file_package_id, import);
+  const auto& [import_package_name, import_library_name] = import_key;
+
+  if (import_package_name == CppPackageName) {
+    if (import_library_name.empty()) {
+      // Clang is not crash-resilient.
+      CARBON_DIAGNOSTIC(CppInteropMissingLibrary, Error,
+                        "`{0}` import missing library", std::string);
+      unit_info.emitter.Emit(import.node_id, CppInteropMissingLibrary,
+                             CppPackageName.str());
```

Comment:
Sorry, noticed you're doing this in a few spots. StringLiteral and std::string are supported; it's only StringRef that's disallowed because it's temporary and may not outlive the diagnostic due to buffering. CppPackageName is a StringLiteral so should work directly.
```suggestion
                        "`{0}` import missing library", llvm::StringLiteral);
      unit_info.emitter.Emit(import.node_id, CppInteropMissingLibrary,
                             CppPackageName);
```

Reply:
Move Cpp to the format string.